Classes/Classes Dealing with Complex Numbers.py

```diff
@@ -1,23 +1,70 @@
 # Python has built-in support for complex number objects: https://docs.python.org/3/c-api/complex.html
-class Complex(complex): # Subclass
-    # The traditional operator overload
+import math
+
+
+class Complex(object):
+    def __init__(self, real, imaginary):
+        self.real = real
+        self.imaginary = imaginary
+
+    def __add__(self, no):
+        a = self.real
+        b = self.imaginary
+        c = no.real
+        d = no.imaginary
+        return Complex(a + c, b + d)
+
+    def __sub__(self, no):
+        a = self.real
+        b = self.imaginary
+        c = no.real
+        d = no.imaginary
+        return Complex(a - c, b - d)
+
+    def __mul__(self, no):
+        a = self.real
+        b = self.imaginary
+        c = no.real
+        d = no.imaginary
+        real_mult = (a * c) - (b * d)
+        imag_mult = (a * d) + (b * c)
+        return Complex(real_mult, imag_mult)
+
+    def __truediv__(self, no):
+        a = self.real
+        b = self.imaginary
+        c = no.real
+        d = no.imaginary
+        real_numerator = a * c + b * d
+        imag_numerator = b * c - a * d
+        denom = c * c + d * d
+        real_div = real_numerator / denom
+        imag_div = imag_numerator / denom
+        return Complex(real_div, imag_div)
+
     def mod(self):
-        return Complex(complex.__abs__(self))
-    
+        a = self.real
+        b = self.imaginary
+        return Complex(math.sqrt(a ** 2 + b ** 2), 0)
+
     def __str__(self):
-        return '{0.real:.2f}{0.imag:+.2f}i'.format(self) # display in A.00+B.00i format
-    
-    def __add__(self, number):
-        return Complex(complex.__add__(self, number))
-    
-    def __sub__(self, number):
-        return Complex(complex.__sub__(self, number))
-    
-    def __mul__(self, number):
-        return Complex(complex.__mul__(self, number))
-    
-    def __truediv__(self, number): # Python3 uses truediv as the name of the '/' operator, whereas python2 just calls this 'div'
-        return Complex(complex.__truediv__(self, number))
-    
-    
-    
+        if self.imaginary == 0:
+            result = '%.2f+0.00i' % (self.real)
+        elif self.real == 0:
+            if self.imaginary >= 0:
+                result = '0.00+%.2fi' % (self.imaginary)
+            else:
+                result = '0.00-%.2fi' % (abs(self.imaginary))
+        elif self.imaginary > 0:
+            result = '%.2f+%.2fi' % (self.real, self.imaginary)
+        else:
+            result = '%.2f-%.2fi' % (self.real, abs(self.imaginary))
+        return result
+
+
+if __name__ == '__main__':
```

Comment:
Remove this section before committing -- it's non-editable in the actual challenge.